src/all/nhentai/src/eu/kanade/tachiyomi/extension/all/nhentai/NHentai.kt

```diff
@@ -255,30 +247,37 @@ open class NHentai(
 
     override fun chapterListSelector() = throw UnsupportedOperationException()
 
-    override fun pageListParse(document: Document) = throw UnsupportedOperationException()
-
-    override fun pageListParse(response: Response): List<Page> {
-        val html = response.body.string()
-
-        val json = dataRegex.find(html)?.groupValues!![1]
-        val data = json.parseAs<Hentai>()
-        val cdnJson = Regex("""image_cdn_urls:\s*(\[.*])""").find(html)?.groupValues!![1]
-        val cdnList = cdnJson.parseAs<List<String>>()
+    override fun pageListParse(document: Document): List<Page> {
+        val data = document.getHentaiData()
+        val cdnList = document.getCDNs()
 
         return data.images.pages.mapIndexed { i, image ->
             Page(
-                i,
-                imageUrl = "https://${cdnList.random()}/galleries/${data.media_id}/${i + 1}" +
-                    when (image.t) {
-                        "w" -> ".webp"
-                        "p" -> ".png"
-                        "g" -> ".gif"
-                        else -> ".jpg"
-                    },
+                index = i,
+                imageUrl = "https://${cdnList.random()}/galleries/${data.media_id}/${i + 1}.${image.extension}",
             )
         }
     }
 
+    private fun Document.getHentaiData(): Hentai {
+        val script = selectFirst(hentaiSelector)!!.data()
+        return dataRegex.find(script)!!.groupValues[1].parseAs()
+    }
+
+    private fun Document.getCDNs(thumbnail: Boolean = false): List<String> {
```

Comment:
Two nitpicks:
1. Should probably use `getCdns` (or `getCdnUrls`)
2. Don't add a default argument value since there are only 2 call sites